src/app/actions/swap.js

```diff
@@ -126,6 +151,50 @@ export const initiateSwap = (swap) => (dispatch, getState) => {
     .catch((e) => dispatch(swapInitFailed(swap.id, e.message || e)))
 }
 
+export const createManualSwap = (swap) => (dispatch, getState) => {
+  let swapId = ''
+  dispatch(swapInitStarted(swap.id))
+  return dispatch(createManualOrder(swap))
+    .then((s) => { 
+      swapId = s.orderId 
+      return dispatch(swapAdded(s))
+    })
+    .then(() => dispatch(addManualProperty(swapId)))
+    .then(() => dispatch(swapInitSuccess(swapId)))
+    .then(() => getSwap(getState(), swapId))
+    .catch((e) => dispatch(swapInitFailed(swapId, e.message || e)))
+}
+
+export const fetchManualSwap = (swapId) => (dispatch, getState) => {
+  return Faast.fetchSwap(swapId)
+    .then((swap) => { 
+      const { swap_id, created_at, user_id, deposit_address, deposit_amount, deposit_currency, 
+        spot_price, price, price_locked_at, price_locked_until, withdrawal_address, 
+        withdrawal_amount, withdrawal_currency, status } = swap
+      const stateSwap = {
```

Comment:
Rather than converting all these fields in the action, use the `formatOrderResult` helper in `Services/Faast`